nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/connect/LauncherRegisterPromptHelper.java

```diff
@@ -0,0 +1,175 @@
+ * @since 8.3
+ */
+public class LauncherRegisterPromptHelper {
+    private static final Log log = LogFactory.getLog(LauncherRegisterPromptHelper.class);
+
+    private static final int PAGE_SIZE = 20;
+
+    public static String promptMail() {
+        return prompt("Mail Adress: ", "^.+@.+\\..{2,4}", "Mail is invalid.");
+    }
+
+    public static String promptUsername() {
+        return prompt("Username: ", "^\\w+", "Username can not be empty.");
+    }
+
+    public static String promptCompany() {
+        return prompt("Company: ", "^\\w+", "Company can't be empty.");
+    }
+
+    public static String promptProjectName() {
+        return prompt("Project name: ", "^(?:[-\\w]+|)$",
+                "Project name can only contain alphanumeric chars and dashes.");
+    }
+
+    public static String promptDescription() {
+        return prompt("Description: ", "", null);
+    }
+
+    protected static String prompt(String message, String regex, String error) {
+        if (StringUtils.isBlank(regex)) {
+            regex = ".*";
+        }
+
+        Console console = System.console();
+        String value = null;
+        do {
+            if (value != null) {
+                console.printf(error + "\n", value);
+            }
+            value = console.readLine(message);
+        } while (!value.matches(regex));
+
+        return value;
+    }
+
+    public static String promptPassword(boolean confirmation) throws IOException, ConfigurationException {
+        String pwd = promptPassword("Please enter your password: ");
+        if (confirmation) {
+            String pwdVerification = promptPassword("Please re-enter your password: ");
+            if (!pwd.equals(pwdVerification)) {
+                throw new ConfigurationException("Passwords do not match.");
+            }
+        }
+        return pwd;
+    }
+
+    public static NuxeoClientInstanceType promptInstanceType() {
+        NuxeoClientInstanceType ret;
+        Console console = System.console();
+
+        do {
+            String s = console.readLine("Instance type (dev|preprod|prod): ");
+            if (StringUtils.isBlank(s)) {
+                ret = NuxeoClientInstanceType.DEV;
+            } else {
+                ret = NuxeoClientInstanceType.fromString(s);
+            }
+        } while (ret == null);
+
+        return ret;
+    }
+
+    public static String promptPassword() throws IOException, ConfigurationException {
+        return promptPassword(false);
+    }
+
+    protected static String promptPassword(String message) throws IOException {
+        Console console = System.console();
+        return new String(console.readPassword(message));
+    }
+
+    public static boolean promptAcceptTerms() {
+        Console console = System.console();
+        if (console != null) {
+            String terms = console.readLine("Read and accept the Nuxeo Trial Terms and Conditions - https://www.nuxeo.com/en/services/nuxeo-trial-tc (Y/n): ");
```

Comment:
Put the URL itself in a constant at the top of the file, we may want to change it separately at some point.